Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -909,28 +920,49 @@ static AffineTransform documentSpaceToPageSpaceTransform(const IntDegrees& pageR
             notifyCursorChanged(toWebCoreCursorType(pdfElementTypes, altKeyIsActive));
             return true;
         }
+        case WebMouseEventButton::Left: {
+            if (m_trackedAnnotation && m_trackedAnnotation != annotationForRootViewPoint(event.position())) {
+                handleMouseDraggedOffTrackedAnnotation();
+                return true;
+            }
+            return false;
+        }
         default:
             return false;
         }
     case WebEventType::MouseDown:
         switch (event.button()) {
         case WebMouseEventButton::Left: {
-            auto pointInDocumentSpace = convertFromPluginToDocument(convertFromRootViewToPlugin(event.position()));
-            auto nearestPageIndex = nearestPageIndexForDocumentPoint(pointInDocumentSpace);
-            if (!nearestPageIndex)
+            if (RetainPtr<PDFAnnotation> annotation = annotationForRootViewPoint(event.position())) {
```

Comment:
Does the coordinate conversion work for iframe PDFs?

Reply:
It looks like it does for the most part, but it seems to breakdown a bit when decreasing the size of iframes. Since this was a refactoring of some already existing logic into a function I'll create a follow up patch for it